ci-operator/step-registry/ipi/conf/aws/user-permissions/ipi-conf-aws-user-permissions-commands.sh

```diff
@@ -0,0 +1,290 @@
+iam:GetUser
+iam:GetUserPolicy
+iam:ListAccessKeys
+iam:ListAttachedRolePolicies
+iam:ListInstanceProfiles
+iam:ListInstanceProfilesForRole
+iam:ListRolePolicies
+iam:ListRoles
+iam:ListUserPolicies
+iam:ListUsers
+iam:PassRole
+iam:PutRolePolicy
+iam:PutUserPolicy
+iam:RemoveRoleFromInstanceProfile
+iam:SimulatePrincipalPolicy
+iam:TagRole
+iam:TagUser
+iam:UntagRole
+route53:ChangeResourceRecordSets
+route53:ChangeTagsForResource
+route53:CreateHostedZone
+route53:DeleteHostedZone
+route53:GetChange
+route53:GetHostedZone
+route53:ListHostedZones
+route53:ListHostedZonesByName
+route53:ListResourceRecordSets
+route53:ListTagsForResource
+route53:UpdateHostedZoneComment
+s3:AbortMultipartUpload
+s3:CreateBucket
+s3:DeleteBucket
+s3:DeleteObject
+s3:GetAccelerateConfiguration
+s3:GetBucketAcl
+s3:GetBucketCors
+s3:GetBucketLocation
+s3:GetBucketLogging
+s3:GetBucketObjectLockConfiguration
+s3:GetBucketPublicAccessBlock
+s3:GetBucketReplication
+s3:GetBucketRequestPayment
+s3:GetBucketTagging
+s3:GetBucketVersioning
+s3:GetBucketWebsite
+s3:GetEncryptionConfiguration
+s3:GetLifecycleConfiguration
+s3:GetObject
+s3:GetObjectAcl
+s3:GetObjectTagging
+s3:GetObjectVersion
+s3:GetReplicationConfiguration
+s3:HeadBucket
+s3:ListBucket
+s3:ListBucketMultipartUploads
+s3:ListBucketVersions
+s3:PutBucketAcl
+s3:PutBucketPublicAccessBlock
+s3:PutBucketTagging
+s3:PutEncryptionConfiguration
+s3:PutLifecycleConfiguration
+s3:PutObject
+s3:PutObjectAcl
+s3:PutObjectTagging
+servicequotas:ListAWSDefaultServiceQuotas
+tag:GetResources
+EOF
+	# additional permisions for 4.11+
+	if ((ocp_minor_version >= 11 && ocp_major_version == 4)); then
+		echo "ec2:DeletePlacementGroup" >>"${PERMISION_LIST}"
+		echo "s3:GetBucketPolicy" >>"${PERMISION_LIST}"
+	fi
+
+	# additional permisions for 4.14+
+	if ((ocp_minor_version >= 14 && ocp_major_version == 4)); then
+		echo "ec2:DescribeSecurityGroupRules" >>"${PERMISION_LIST}"
+
+		# sts:AssumeRole is required for Shared-VPC install https://issues.redhat.com/browse/OCPBUGS-17751
+		echo "sts:AssumeRole" >>"${PERMISION_LIST}"
+	fi
+
+	# additional permisions for 4.15+
+	if ((ocp_minor_version >= 15 && ocp_major_version == 4)); then
+		echo "iam:TagInstanceProfile" >>"${PERMISION_LIST}"
+	fi
+
+	# additional permisions for 4.16+
+	if ((ocp_minor_version >= 16 && ocp_major_version == 4)); then
+		echo "ec2:DisassociateAddress" >>"${PERMISION_LIST}"
+		echo "elasticloadbalancing:SetSecurityGroups" >>"${PERMISION_LIST}"
+		echo "s3:PutBucketPolicy" >>"${PERMISION_LIST}"
+	fi
+
+	# generate policy file
+	PERMISION_JSON="${ARTIFACT_DIR}/permision_list.json"
+	jq --raw-input . "${PERMISION_LIST}" | jq -sc >"${PERMISION_JSON}"
+
+	cat <<EOF >"${USER_POLICY_FILE}"
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [],
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+	cat <<<"$(jq '.Statement[0].Action += input' "${USER_POLICY_FILE}" "${PERMISION_JSON}")" >"${USER_POLICY_FILE}"
```

Comment:
Or maybe we use python to eliminate the noise of download `jq` from github, something like:
```
tmp_user_policy_file=$(mktemp)
cat "${USER_POLICY_FILE}" | python3 -c 'import json,sys; j=json.load(sys.stdin);data=open("'${PERMISION_LIST}'", "r").read().splitlines();j["Statement"][0]["Action"]+=data;print(json.dumps(j))' > "${tmp_user_policy_file}"
mv "${tmp_user_policy_file}" "${USER_POLICY_FILE}"
...

Reply:
I've replaced all jq invocations with a python script.